### ISSUE TITLE
Add example toggle to switch between shared/static library build

### ISSUE THREAD

**Issue by MartinHelmut** [MartinHelmut/cpp-gui-template-sdl2#4]
## Description

Even the improved version of the template does not show how to easily toggle between building shared or static libs, taking packaging into account. A project option via CMake would be amazing to showcase this.

**Comment by pusewicz** [comment]
@MartinHelmut Yep, it would be awesome. Especially with the weird packaging for the macOS. Not to mention when dealing with hot-reloading in development.
